jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/util/DemandingFlusher.java

```diff
@@ -41,7 +42,7 @@ public abstract class DemandingFlusher extends IteratingCallback implements Dema
     private static final Throwable SENTINEL_CLOSE_EXCEPTION = new StaticException("Closed");
 
     private final IncomingFrames _emitFrame;
-    private final AtomicLong _demand = new AtomicLong();
+    private final AtomicBoolean _demand = new AtomicBoolean();
```

Comment:
Why a tri-state that was necessary before is not needed now?

Reply:
Well it was a long before, so it was like this before just in a different way.

demand>0 was for demanding
demand==0 was for no demand
demand<0 was for cancelled demand